repository/src/main/java/org/apache/atlas/accesscontrol/AccessControlUtil.java

```diff
@@ -0,0 +1,388 @@
+        int size = 25;
+        int from = 0;
+
+        params.put("policyLabelsPartial", label);
+        params.put("page", "0");
+        params.put("pageSize", String.valueOf(size));
+
+        if (StringUtils.isNotEmpty(serviceType)) {
+            params.put("serviceType", serviceType);
+        }
+
+        if (StringUtils.isNotEmpty(policyType)) {
+            params.put("policyType", policyType);
+        }
+
+        int fetched;
+        do {
+            params.put("startIndex", String.valueOf(from));
+
+            List<RangerPolicy> rangerPolicies = atlasRangerService.getPoliciesByLabel(params);
+            fetched = rangerPolicies.size();
+            ret.addAll(rangerPolicies);
+
+            from += size;
+
+        } while (fetched == size);
+
+        return ret;
+    }
+
+    public static Map<String, Object> mapOf(String key, Object value) {
+        Map<String, Object> map = new HashMap<>();
+        map.put(key, value);
+
+        return map;
+    }
+
+    public static void ensureNonAccessControlEntityType(List<String> types) throws AtlasBaseException {
+        AtlasPerfMetrics.MetricRecorder metricRecorder = RequestContext.get().startMetricRecord("ensureNonAccessControlEntityType");
+        long accessControlEntityCount = types.stream().filter(ACCESS_CONTROL_ENTITY_TYPES::contains).count();
+
+        if (accessControlEntityCount > 0) {
+            throw new AtlasBaseException(OPERATION_NOT_SUPPORTED);
+        }
+
+        RequestContext.get().endMetricRecord(metricRecorder);
+    }
+
+    public static void ensureNonAccessControlRelType(String type) throws AtlasBaseException {
+        AtlasPerfMetrics.MetricRecorder metricRecorder = RequestContext.get().startMetricRecord("ensureNonAccessControlRelType");
+        if (ACCESS_CONTROL_RELATION_TYPE.equals(type)) {
+            throw new AtlasBaseException(OPERATION_NOT_SUPPORTED);
+        }
+
+        RequestContext.get().endMetricRecord(metricRecorder);
+    }
+
+    public static ExecutorService getExecutorService(int numThreads, String threadNamePattern) {
+        ExecutorService service = Executors.newFixedThreadPool(numThreads, new ThreadFactoryBuilder().setNameFormat(threadNamePattern + Thread.currentThread().getName()).build());
+        return service;
+    }
+
+    public static <T> void submitCallablesAndWaitToFinish(String threadName, List<Callable<T>> callables) throws AtlasBaseException {
+        ExecutorService service = getExecutorService(callables.size(), threadName + "-%d-");
+        try {
+            callables.forEach(service::submit);
+
+            service.shutdown();
+            LOG.info("Shutting down executor: {}", threadName);
+            boolean terminated = service.awaitTermination(60, TimeUnit.SECONDS);
+            LOG.info("Shut down executor: {}", threadName);
+
+            if (!terminated) {
+                LOG.warn("Time out occurred while waiting to complete {}", threadName);
```

Comment:
Should not we throw exception when the service does not terminates properly in the defined time interval?

Reply:
no termination within timeout is not always a failure, not sure about throwing exception here